src/SymbolCollector.Core/Client.cs

```diff
@@ -107,62 +112,118 @@ private async Task UploadAsync(string debugId, string file, CancellationToken ca
         // TODO: IAsyncEnumerable when ELF library supports it
         private IEnumerable<(string debugId, string file)> GetFiles(IEnumerable<string> files)
         {
+            Func<string, string?> getBuildId;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                // TODO: Take this as a strategy class-wide to check for platform only once.
+                getBuildId = GetMachOBuildId;
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                getBuildId = GetElfBuildId;
+            }
+            else
+            {
+                // TODO: This needs to be check at the start of the program though
+                throw new NotSupportedException($"OS {RuntimeInformation.OSDescription} is not supported.");
+            }
+
             foreach (var file in files)
             {
                 _logger.LogInformation("Processing file: {file}.", file);
-                IELF? elf = null;
-                string? buildIdHex;
-                try
-                {
-                    // TODO: find an async API
-                    if (!ELFReader.TryLoad(file, out elf))
-                    {
-                        _logger.LogWarning("Couldn't load': {file} with ELF reader.", file);
-                        continue;
-                    }
 
-                    var hasBuildId = elf.TryGetSection(".note.gnu.build-id", out var buildId);
-                    if (!hasBuildId)
-                    {
-                        _logger.LogWarning("No Debug Id in {file}", file);
-                        continue;
-                    }
+                var buildId = getBuildId(file);
+                if (buildId is null) continue;
 
-                    var hasUnwindingInfo = elf.TryGetSection(".eh_frame", out _);
-                    var hasDwarfDebugInfo = elf.TryGetSection(".debug_frame", out _);
+                yield return (buildId, file);
+            }
+        }
 
-                    if (!hasUnwindingInfo && !hasDwarfDebugInfo)
+        private string? GetMachOBuildId(string file)
+        {
+            try
+            {
+                // TODO: find an async API if this is used by the server
+                if (MachOReader.TryLoad(file, out _) == MachOResult.OK)
+                {
+                    using var algorithm = SHA256.Create();
+                    var hash = algorithm.ComputeHash(File.ReadAllBytes(file));
```

Comment:
The Mach header specifies a UUID. Hashing the file is not necessarily going to create the correct id.

Reply:
Right the hash was just some way to key this and open a PR.
I need to figure out a way to grab this UUID then.

Reply:
https://github.com/getsentry/symbolic/blob/46d0c5f794ed89b0c3adcd5d362259f876054938/debuginfo/src/macho.rs#L52-L60